ShopAppTests/App/Account/ViewModels/SettingsViewModelSpec.swift

```diff
@@ -0,0 +1,95 @@
+//
+//  SettingsViewModelSpec.swift
+//  ShopAppTests
+//
+//  Created by Radyslav Krechet on 2/28/18.
+//  Copyright © 2018 RubyGarage. All rights reserved.
+//
+
+import Nimble
+import Quick
+import RxSwift
+
+@testable import ShopApp
+
+class SettingsViewModelSpec: QuickSpec {
+    override func spec() {
+        let repositoryMock = AuthentificationRepositoryMock()
+        let updateCustomerUseCaseMock = UpdateCustomerUseCaseMock(repository: repositoryMock)
+        let loginUseCaseMock = LoginUseCaseMock(repository: repositoryMock)
+        let customerUseCaseMock = CustomerUseCaseMock(repository: repositoryMock)
+        
+        var viewModel: SettingsViewModel!
+        
+        beforeEach {
+            viewModel = SettingsViewModel(updateCustomerUseCase: updateCustomerUseCaseMock, loginUseCase: loginUseCaseMock, customerUseCase: customerUseCaseMock)
+        }
+        
+        describe("when view model initialized") {
+            it("should have variable with correct initial value") {
+                expect(viewModel.customer.value).to(beNil())
+            }
+        }
+        
+        describe("when customer loaded") {
+            context("if it have success customer load") {
+                it("should have variables with a correct values") {
+                    customerUseCaseMock.isNeedToReturnError = false
+                    viewModel.loadCustomer()
+                    
+                    expect(viewModel.customer.value).toNot(beNil())
+                }
+            }
+            
+            context("if it have fail customer load") {
+                it("should have variables with a correct values") {
+                    customerUseCaseMock.isNeedToReturnError = true
+                    viewModel.loadCustomer()
+                    
+                    expect(viewModel.customer.value).to(beNil())
+                }
+            }
+        }
+        
+        describe("when settings changed") {
+            var disposeBag: DisposeBag!
+
+            beforeEach {
+                disposeBag = DisposeBag()
+                
+                customerUseCaseMock.isNeedToReturnError = false
+                viewModel.loadCustomer()
+            }
+            
+            context("and it successed") {
+                it("needs to show content of screen") {
+                    updateCustomerUseCaseMock.isNeedToReturnError = true
+                    viewModel.setPromo(true)
+                    
+                    expect(viewModel.customer.value?.promo) == true
+                    
+                    viewModel.state
```

Comment:
Can we verify state after calling `viewModel.setPromo(true)` and before `updateCustomerUseCaseMock` result to be sure that state is .loading in this moment?